hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -524,16 +524,16 @@ def b64decode(s):
 
 _cases = []  # type: list
 
-
-def bad_django_TestCase(runner):
-    if runner is None:
-        return False
-    if not _cases:
-        try:
-            from django.test import TransactionTestCase
-            from hypothesis.extra.django import HypothesisTestCase
-            _cases[:] = TransactionTestCase, HypothesisTestCase
-        except Exception:
-            # Can't use ImportError, because of e.g. Django config errors
-            _cases[:] = (), type
-    return isinstance(runner, _cases[0]) and not isinstance(runner, _cases[1])
+try:
+    from django.test import TransactionTestCase
+    from hypothesis.extra.django import HypothesisTestCase
+
+    def bad_django_TestCase(runner):
+        if runner is None:
+            return False
+        return isinstance(runner, TransactionTestCase) and \
+            not isinstance(runner, HypothesisTestCase)
+except Exception:
+    # Can't use ImportError, because of e.g. Django config errors
+    def bad_django_TestCase(runner):
+        pass
```

Comment:
I'd suggest `return False` here. (It's equivalent in our use, but makes the intent much clearer.)

(No time for a full review at the moment, but I'm following the thread I started so figured I'd leave a drive-by comment...)

Reply:
Good point. Will change.

Reply:
`return False` here.

Reply:
Whoops. I did fix it, but apparently that got lost in some rebase shenanigans.